pkg/gameserverallocations/controller.go

```diff
@@ -254,119 +243,140 @@ func (c *Controller) loggerForGameServerAllocation(gsa *v1alpha1.GameServerAlloc
 	return c.loggerForGameServerAllocationKey(gsaName).WithField("gsa", gsa)
 }
 
-// creationMutationHandler will intercept when a GameServerAllocation is created, and allocate it a GameServer
-// assuming that one is available. If not, it will reject the AdmissionReview.
-func (c *Controller) creationMutationHandler(review admv1beta1.AdmissionReview) (admv1beta1.AdmissionReview, error) {
-	c.baseLogger.WithField("review", review).Info("creationMutationHandler")
-	obj := review.Request.Object
-	gsa := &v1alpha1.GameServerAllocation{}
+// allocationHandler CRDHandler for allocating a gameserver. Only accepts POST
+// commands
+func (c *Controller) allocationHandler(w http.ResponseWriter, r *http.Request, namespace string) error {
```

Comment:
/cc @pooneh-m @ilkercelikyilmaz for your review and testing (also, I don't think I broke anything you did @ilkercelikyilmaz , but would love for you to give it a look over to make sure) :+1:

Reply:
LGTM